src/Products/CMFCore/explicitacquisition.py

```diff
@@ -0,0 +1,23 @@
+from zExceptions import NotFound
+from zope.component import adapter
+from ZPublisher.interfaces import IPubAfterTraversal
+
+from Products.CMFCore.interfaces import IContentish
+from Products.CMFCore.interfaces import IPublishableThroughAcquisition
+from Products.CMFCore.interfaces import IShouldAllowAcquiredItemPublication
+
+
+@adapter(IPubAfterTraversal)
+def after_traversal_hook(event):
+    if IPublishableThroughAcquisition.providedBy(event.request):
```

Comment:
I would add an environment variable to control the behavior, because in the rare case anybody depends on the (broken) old behavior it is easier to work around. 
Like so (needs test if it works ... and docs) and also probably below too: 
```suggestion
import os

PUBLISHABLE_THROUGH_ACQUISITION =  os.environ.get("ALLOW_CONTENT_PUBLISHABLE_THROUGH_ACQUISITION", "false") == "true"

@adapter(IPubAfterTraversal)
def after_traversal_hook(event):
    if  PUBLISHABLE_THROUGH_ACQUISITION or IPublishableThroughAcquisition.providedBy(event.request):
```